Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/BayesStretch2Test.py

```diff
@@ -0,0 +1,331 @@
+        self._alg.initialize()
+        self._N_hist = 1
+
+    def tearDown(self):
+        """
+        Remove workspaces from ADS.
+        """
+        DeleteWorkspace(self._sample_ws)
+        DeleteWorkspace(self._res_ws)
+
+    def assert_mock_called_with(self, mock_object, N_calls, call_number, **kargs):
+        self.assertEqual(N_calls, mock_object.call_count)
+
+        called_with = mock_object.call_args_list[call_number - 1][1]  # dict of {keyword: value}
+        self.assertEqual(len(kargs), len(called_with))
+        for j, keyword in enumerate(kargs):
+            expected = kargs[keyword]
+            if isinstance(expected, np.ndarray):
+                np.testing.assert_array_equal(expected, called_with[keyword])
+            else:
+                self.assertEqual(expected, called_with[keyword])
+
+    def point_mock(self, name):
+        if "res" in name:
+            return self._res_ws, self._N_hist
+        else:
+            return self._sample_ws, self._N_hist
+
+    def test_make_contour(self):
+        x = np.linspace(0.01, 0.1, 3)
+        y = np.linspace(0.8, 1.0, 3)
+        X, Y = np.meshgrid(x, y)
+
+        Z = np.sin(X) - np.cos(Y)
+
+        ws_str = self._alg.make_contour(X, Y, Z, 0, "test")
+        ws = AnalysisDataService.retrieve(ws_str)
+        expect = [[-0.687, -0.642, -0.597], [-0.612, -0.567, -0.522], [-0.530, -0.485, -0.440]]
+
+        self.assertEqual(ws.getNumberHistograms(), 3)
+        for j in range(ws.getNumberHistograms()):
+            np.testing.assert_almost_equal(ws.readX(j), x, 3)
+            np.testing.assert_almost_equal(ws.readY(j), expect[j], 3)
+
+        label = ["beta", "FWHM"]
+        axis_values = [x, y]
+        for i in range(ws.axes()):
+            axis = ws.getAxis(i)
+            unit = axis.getUnit()
+            self.assertEqual(unit.caption(), label[i])
+            np.testing.assert_equal(axis.extractValues(), axis_values[i])
+
+        DeleteWorkspace(ws)
+
+    def test_make_slice_ws(self):
+        x = np.linspace(0.8, 1.0, 3)
+        y1 = -((x - 0.9) ** 2) + 0.01
+        y2 = -((x - 0.86) ** 2) + 0.01
+
+        Q = [0.4, 0.8]
+        slices = [(x, y1), (x, y2)]
+
+        ws_str = self._alg.make_slice_ws(slices, Q, "MomentumTransfer", "test")
+
+        ws = AnalysisDataService.retrieve(ws_str)
+        expect = [[0, 1e-2, 0.0], [0.006, 0.008, -0.01]]
+
+        self.assertEqual(ws.getNumberHistograms(), 2)
+        for j in range(ws.getNumberHistograms()):
+            np.testing.assert_almost_equal(ws.readX(j), x, 3)
+            np.testing.assert_almost_equal(ws.readY(j), expect[j], 3)
+
+        label = ["q", ""]
+        axis_values = [x, [str(val) for val in Q]]
+        for i in range(ws.axes()):
+            axis = ws.getAxis(i)
+            unit = axis.getUnit()
+            self.assertEqual(unit.caption(), label[i])
+            np.testing.assert_equal(axis.extractValues(), axis_values[i])
+        DeleteWorkspace(ws)
```

Comment:
Same comment here. We can clear the ADS when we tearDown, and so wouldn't need the DeleteWorkspace algo

Reply:
And I think the same for the `BayesQuasi2Test`